src/state/internal/createQueryStore.ts

```diff
@@ -0,0 +1,999 @@
+    params?: { [K in keyof TParams]: ParamResolvable<TParams[K], TParams, StoreState<TData, TParams> & PrivateStoreState & U, TData> };
+  },
+  arg1?:
+    | StateCreator<U, [], [['zustand/subscribeWithSelector', never]]>
+    | RainbowPersistConfig<StoreState<TData, TParams> & PrivateStoreState & U>,
+  arg2?: RainbowPersistConfig<StoreState<TData, TParams> & PrivateStoreState & U>
+): QueryStore<TData, TParams, StoreState<TData, TParams> & U> {
+  type S = StoreState<TData, TParams> & PrivateStoreState & U;
+
+  /* If arg1 is a function, it's the customStateCreator; otherwise, it's the persistConfig. */
+  const customStateCreator = typeof arg1 === 'function' ? arg1 : () => ({}) as U;
+  const persistConfig = typeof arg1 === 'object' ? arg1 : arg2;
+
+  const {
+    abortInterruptedFetches = true,
+    fetcher,
+    onFetched,
+    transform,
+    cacheTime = time.days(7),
+    debugMode = false,
+    disableAutoRefetching = false,
+    disableCache = false,
+    enabled = true,
+    keepPreviousData = false,
+    maxRetries = 3,
+    onError,
+    params,
+    retryDelay = time.seconds(5),
+    setData,
+    staleTime = time.minutes(2),
+    suppressStaleTimeWarning = false,
+  } = config;
+
+  if (IS_DEV && !suppressStaleTimeWarning && staleTime < MIN_STALE_TIME) {
+    console.warn(
+      `[createQueryStore${persistConfig?.storageKey ? `: ${persistConfig.storageKey}` : ''}] ❌ Stale times under ${
+        MIN_STALE_TIME / 1000
+      } seconds are not recommended.`
+    );
+  }
+
+  let directValues: Partial<TParams> | null = null;
+  let paramAttachVals: Partial<Record<keyof TParams, AttachValue<unknown>>> | null = null;
+  let paramUnsubscribes: Unsubscribe[] = [];
+  let fetchAfterParamCreation = false;
+
+  let activeAbortController: AbortController | null = null;
+  let activeFetch: { key: string; promise?: Promise<TData | null> } | null = null;
+  let activeRefetchTimeout: NodeJS.Timeout | null = null;
+  let lastFetchKey: string | null = null;
+
+  const enableLogs = IS_DEV && debugMode;
+
+  const initialData = {
+    enabled,
+    error: null,
+    lastFetchedAt: null,
+    queryCache: {},
+    queryKey: '',
+    status: QueryStatuses.Idle,
+    subscriptionCount: 0,
+  };
+
+  const getQueryKey = (params: TParams): string => JSON.stringify(Object.values(params));
+
+  const getCurrentResolvedParams = () => {
+    const currentParams: Partial<TParams> = directValues ? { ...directValues } : {};
+    for (const k in paramAttachVals) {
+      const attachVal = paramAttachVals[k as keyof TParams];
+      if (!attachVal) continue;
+      currentParams[k as keyof TParams] = attachVal.value as TParams[keyof TParams];
+    }
+    return currentParams as TParams;
+  };
+
+  const fetchWithAbortControl = async (params: TParams): Promise<TQueryFnData> => {
+    const abortController = new AbortController();
+    activeAbortController = abortController;
```

Comment:
any reason to create a local variable here instead of just assigning directly to `activeAbortController`?

Reply:
Used here to make sure that `activeAbortController` is only nulled out if it hasn't been reassigned:

https://github.com/rainbow-me/rainbow/blob/f0d4737391845e909add9eea0beec05e34b22a28/src/state/internal/createQueryStore.ts#L486-L490